client/metric.go

```diff
@@ -0,0 +1,64 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/drand/drand/chain"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func newMetricController(chainInfo *chain.Info, r prometheus.Registerer) *metricController {
+	return &metricController{chainInfo: chainInfo, bridge: r}
+}
+
+type metricController struct {
+	chainInfo *chain.Info
+	bridge    prometheus.Registerer
+}
+
+func (mc *metricController) Register(x prometheus.Collector) error {
+	return mc.bridge.Register(x)
+}
+
+func newWatchLatencyMetricClient(id string, base Client, ctl *metricController) (Client, error) {
+	c := &watchLatencyMetricClient{
+		Client: base,
+		ctl:    ctl,
+		watchLatency: prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
Probably want a single [`GaugeVec`](https://godoc.org/github.com/prometheus/client_golang/prometheus#NewGaugeVec) that specifies a label on the id of the specific watcher